src/Core/Services/Implementations/SendService.cs

```diff
@@ -43,6 +51,25 @@ public class SendService : ISendService
 
         public async Task SaveSendAsync(Send send)
         {
+            // Make sure user can save Sends
+            if (send.UserId.HasValue)
+            {
+                var policies = await _policyRepository.GetManyByUserIdAsync(send.UserId.Value);
+                if (policies != null)
+                {
+                    foreach (var policy in policies.Where(p => p.Enabled && p.Type == PolicyType.DisableSend))
+                    {
+                        var org = await _organizationUserRepository.GetDetailsByUserAsync(send.UserId.Value, policy.OrganizationId,
+                            OrganizationUserStatusType.Confirmed);
+                        if (org != null && org.Enabled && org.UsePolicies
+                           && org.Type != OrganizationUserType.Admin && org.Type != OrganizationUserType.Owner)
+                        {
+                            throw new BadRequestException("Due to an Enterprise Policy, you are restricted to only Deleting Sends.");
+                        }
+                    }
+                }
+            }
```

Comment:
is it possible to unnest some of these conditions and just return? i feel like it would read easier.